buildspec.yml

```diff
@@ -42,65 +42,76 @@ phases:
       - cp -r docker/build_artifacts/* $build_dir/
       - CPU_PY2_TAG="$FRAMEWORK_VERSION-cpu-py2-$BUILD_ID"
       - GPU_PY2_TAG="$FRAMEWORK_VERSION-gpu-py2-$BUILD_ID"
-      - docker build -f "$build_dir/$cpu_dockerfile" -t $PREPROD_IMAGE:$CPU_PY2_TAG $build_dir
-      - docker build -f "$build_dir/$gpu_dockerfile" -t $PREPROD_IMAGE:$GPU_PY2_TAG $build_dir
+      - build_cmd="docker build -f "$build_dir/$cpu_dockerfile" -t $PREPROD_IMAGE:$CPU_PY2_TAG $build_dir"
+      - execute-command-if-has-matching-changes "$build_cmd" "test/" "tests/" "docker/*" "buildspec.yml"
```

Comment:
not strictly necessary but it might be good to make `'"test/" "tests/" "docker/*" "buildspec.yml"'` a constant since it's getting repeated everywhere

Reply:
i like this idea, will do with a next pass on making those tests parallel.